.github/workflows/test-and-build-docker-images.yml

```diff
@@ -237,22 +237,21 @@ jobs:
         name: image-digests
         path: ${{ env.IMAGE_DIGEST_FILENAME }}
 
-    - name: Sign container images
-      if: inputs.should-push-image == 'true'
-      env:
-        TAGS: ${{ inputs.tags }}
-      run: |
-        for tag in $TAGS
-        do 
-          cosign sign $tag
-        done
-
-    - name: Verify container images
-      if: inputs.should-push-image == 'true'
+    - name: Sign and verify container images
+      if: inputs.should-push-image == 'true' && inputs.release-build == 'true'
       env:
         TAGS: ${{ inputs.tags }}
+        IMAGE_NAME: ${{ inputs.artifact }}
+        IMAGE_DIGEST: ${{ steps.docker_build_image.outputs.digest }}
       run: |
+        # cosign sign $IMAGE_NAME:$IMAGE_DIGEST
+        echo $TAGS
         for tag in $TAGS
         do
-          cosign verify $tag
+          REMOVE=:${tag##*:} # remove everything from the last ":"
+          IMAGE_TO_SIGN=${tag%$REMOVE}@$IMAGE_DIGEST
+          echo "Signing $IMAGE_TO_SIGN"
+          cosign sign --yes $IMAGE_TO_SIGN
+          echo "Verify identity"
+          cosign verify --certificate-identity-regexp="https://github.com/keptn/keptn/.*" --certificate-oidc-issuer="https://token.actions.githubusercontent.com" $IMAGE_TO_SIGN
```

Comment:
The certificate identity used by cosign is the path to the workflow @ the branch ref. Using this regexp we can verify the signature is coming from a build of keptn/keptn